### ISSUE TITLE
Custom profiles for ipyparallel for each controller type

### ISSUE THREAD

**Issue by isaacovercast** [dereneaton/ipyrad#22]
It could be useful to use our own custom profiles rather than relying on the default for ipyparallel. 

Context from gitter: "We could create ipyparallel profiles for each cluster type (local, mpi, pbs), include these profiles in the ipyrad distribution, and use the --profile-dir and --profile flags for all cluster commands to just talk to these profiles? Then it will be very simple to change controller type w/in an assembly object, by just initializing the clients with the appropriate profile."

**Comment by dereneaton** [comment]
Could be. I've read the ipyparallel docs about 5 times, but I haven't actually tried using it to submit jobs on HPC yet. We'll definitely have to do some testing.

On another note, I'm worried it might be difficult to properly split up the `threaded_view` objects onto different nodes when running, say, an HPC connection to 4 nodes with 8 CPUs each. Ideally for running vsearch this would send 1 file to each node running on 8 threads, or 2 files to each running 4 threads, but I don't know how to do that yet. Right now I think we have it to simply see 32 threads and divide it into 8 4-threaded Clients, but I doubt that will actually work if the Client is pulling Engines from different nodes. Maybe I'm wrong, though. Again, needs testing.

**Comment by dereneaton** [comment]
Notes on the state of things: 

The idea of creating profiles is still open. But for now, users can connect different Assemblies to different profiles if they want. I just did this when I had two different notebooks running at once. The first notebook had an Assembly running on the 'default' profile, so for the second one I started a separate ipcluster with profile name "ipyrad" and 4 Engines and was able to get the Assembly object from the second notebook to attach to it by simply setting that ipprofile attribute:
` data.ipprofile = "ipyrad"`

It crashes with an error if the profile doesn't exist. 

User's can also set a `cluster_id` for an Asesmbly object but that is really only used in the CLI to attach assemblies to the 'default' profile and allow shutting down an individual Assembly's connection without killing the controller.
